tests/python/FileTransformTest.py

```diff
@@ -0,0 +1,207 @@
+                       ('houdini', 'lut'),
+                       ('International Color Consortium profile', 'icc'),
+                       ('Image Color Matching profile', 'icm'),
+                       ('ICC profile', 'pf'),
+                       ('iridas_cube', 'cube'),
+                       ('iridas_itx', 'itx'),
+                       ('iridas_look', 'look'),
+                       ('pandora_mga', 'mga'),
+                       ('pandora_m3d', 'm3d'),
+                       ('resolve_cube', 'cube'),
+                       ('spi1d', 'spi1d'),
+                       ('spi3d', 'spi3d'),
+                       ('spimtx', 'spimtx'),
+                       ('truelight', 'cub'),
+                       ('nukevf', 'vf')]
+
+    def setUp(self):
+        self.file_tr = OCIO.FileTransform()
+
+    def tearDown(self):
+        self.file_tr = None
+
+    def test_cccid(self):
+        """
+        Test the setCCCId() and getCCCId() methods.
+        """
+
+        # Default initialized id value is ""
+        self.assertEqual(self.file_tr.getCCCId(), '')
+
+        # Test id name setter and getter.
+        for id_ in TEST_NAMES:
+            self.file_tr.setCCCId(id_)
+            self.assertEqual(self.file_tr.getCCCId(), id_)
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.file_tr.setCCCId(invalid)
+
+    def test_cdlstyle(self):
+        """
+        Test the setCDLStyle() and getCDLStyle() methods.
+        """
+
+        # Default initialized CDL Style value is no clamp.
+        self.assertEqual(self.file_tr.getCDLStyle(),
+                         OCIO.CDLStyle.CDL_NO_CLAMP)
+
+        for cdl_style in OCIO.CDLStyle.__members__.values():
+            self.file_tr.setCDLStyle(cdl_style)
+            self.assertEqual(self.file_tr.getCDLStyle(), cdl_style)
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.file_tr.setCDLStyle(invalid)
+
+    def test_direction(self):
+        """
+        Test the setDirection() and getDirection() methods.
+        """
+
+        # Default initialized direction is forward.
+        self.assertEqual(self.file_tr.getDirection(),
+                         OCIO.TRANSFORM_DIR_FORWARD)
+
+        for direction in OCIO.TransformDirection.__members__.values():
+            # Setting the unknown direction preserves the current direction.
+            if direction != OCIO.TRANSFORM_DIR_UNKNOWN:
+                self.file_tr.setDirection(direction)
+                self.assertEqual(self.file_tr.getDirection(), direction)
+
+        # Wrong type tests.
+        for invalid in (None, 1):
+            with self.assertRaises(TypeError):
+                self.file_tr.setDirection(invalid)
+
+    def test_getformats(self):
+        """
+        Test the getFormats() method.
+        """
+
+        format_iterator = self.file_tr.getFormats()
```

Comment:
``getFormats`` is static, so would be better to test like:
```suggestion
        format_iterator = OCIO.FileTransform.getFormats()
```

Reply:
Addressed in 2d14c6c.